src/super_gradients/training/transforms/transforms.py

```diff
@@ -690,74 +703,84 @@ def __init__(
         self.maybe_flip = DetectionHorizontalFlip(prob=flip_prob)
 
     def close(self):
-        self.additional_samples_count = 0
         self.enable_mixup = False
 
-    def apply_to_sample(self, sample: DetectionSample) -> DetectionSample:
-        if self.enable_mixup and random.random() < self.prob:
-            (cp_sample,) = sample.additional_samples
-            target_dim = self.input_dim if self.input_dim is not None else sample.image.shape[:2]
-
-            cp_sample = self.maybe_flip.apply_to_sample(cp_sample)
-
-            jit_factor = random.uniform(*self.mixup_scale)
+    @property
+    def additional_samples_count(self) -> int:
+        do_mixup = self.enable_mixup and random.random() < self.prob
+        return int(do_mixup)
 
-            if len(sample.image.shape) == 3:
-                cp_img = np.ones((target_dim[0], target_dim[1], sample.image.shape[2]), dtype=np.uint8) * self.border_value
-            else:
-                cp_img = np.ones(target_dim, dtype=np.uint8) * self.border_value
+    @additional_samples_count.setter
+    def additional_samples_count(self, value):
+        pass  # ignored
```

Comment:
Maybe let's throw an exception to explicitly disable writing to this property?

Reply:
Done, [49f5548](https://github.com/Deci-AI/super-gradients/pull/1897/commits/49f55484f022f5938ef472a5eed5900a26fe9d2b)